raiden/tests/unit/transfer/test_node.py

```diff
@@ -56,3 +114,331 @@ def test_is_transaction_effect_satisfied(
+            block_number=chain_state.block_number,
+            gas_limit=GAS_LIMIT,
+            block_hash=chain_state.block_hash,
+        )
+        subdispatch_by_canonical_id(
+            chain_state=chain_state,
+            canonical_identifier=canonical_identifier,
+            state_change=state_change,
+        )
+
+    state_change = ActionChannelClose(canonical_identifier=canonical_identifier)
+
+    # dispatching for an unknown canonical_identifier will not emit events
+    transition_result = subdispatch_by_canonical_id(
+        chain_state=chain_state,
+        canonical_identifier=CanonicalIdentifier(
+            chain_identifier=chain_state.chain_id,
+            token_network_address=factories.make_address(),
+            channel_identifier=factories.make_channel_identifier(),
+        ),
+        state_change=state_change,
+    )
+    assert not transition_result.events, transition_result
+
+    assert get_status(channel_state) == CHANNEL_STATE_OPENED
+    transition_result = subdispatch_by_canonical_id(
+        chain_state=chain_state,
+        canonical_identifier=canonical_identifier,
+        state_change=state_change,
+    )
+
+    assert get_status(channel_state) != CHANNEL_STATE_OPENED
+    assert transition_result.new_state == chain_state, transition_result
+
+
+def test_handle_node_change_network_state(chain_state, netting_channel_state, monkeypatch):
+    state_change = ActionChangeNodeNetworkState(
+        node_address=factories.make_address(), network_state=NODE_NETWORK_REACHABLE
+    )
+    transition_result = handle_node_change_network_state(chain_state, state_change)
+    # no events if no mediator tasks are there to apply to
+    assert not transition_result.events
+
+    mediator_state = MediatorTransferState(
+        secrethash=UNIT_SECRETHASH,
+        routes=[
+            RouteState(
+                route=[netting_channel_state.partner_state.address],
+                forward_channel_id=netting_channel_state.canonical_identifier.channel_identifier,
+            )
+        ],
+    )
+    subtask = MediatorTask(
+        token_network_address=netting_channel_state.canonical_identifier.token_network_address,
+        mediator_state=mediator_state,
+    )
+    chain_state.payment_mapping.secrethashes_to_task[UNIT_SECRETHASH] = subtask
+
+    lock = factories.HashTimeLockState(amount=0, expiration=2, secrethash=UNIT_SECRETHASH)
+
+    netting_channel_state.partner_state.secrethashes_to_lockedlocks[UNIT_SECRETHASH] = lock
+    netting_channel_state.partner_state.merkletree = compute_merkletree_with(
+        merkletree=make_empty_merkle_tree(), lockhash=lock.lockhash
+    )
+    result = object()
+    monkeypatch.setattr(
+        raiden.transfer.node,
+        "subdispatch_mediatortask",
+        lambda *args, **kwargs: TransitionResult(chain_state, [result]),
+    )
+    transition_result = handle_node_change_network_state(chain_state, state_change)
+
+    assert transition_result.events == [result]
```

Comment:
I don't see the point of this test to be honest. So you're putting together a mediator state, then you calling `node.handle_node_change_network_state` which is going to call `subdispatch_mediatortask` and that is a patched function which will return a result that you later on check against. Why don't we let the state change pass all the way down to the mediator?

Reply:
This intends to test the dispatch behavior of `handle_node_change_network_state` unit, not the mediator state machine. An alternative implementation could have been a `MagicMock`, where I assert on `subdispatch_mediatortask` having been called (a certain number of times).
If I don't patch the function call, then testing this unit is dependent on the behavior on the other units.

I can change it, but I don't think it will improve the test. Do you want me to?

Reply:
Ok i think i see what you were testing now. No need to change this then